Debugger.py

```diff
@@ -73,42 +74,83 @@ def parseCommandLineArgs():
                         help="This is required if -s option was used on the linker. That will allow "
                              "binary to be loader at correct address specified inside the binary")
 
+    parser.add_argument("-bp", "--breakpoint",
+                        required=False,
+                        type=str,
+                        default=None,
+                        help="This is an optional argument. If present, debugger will load breakpoints "
+                        "previously defined and stored in the specified file.")
+
     args = parser.parse_args()
 
     return args
 
 
-def validatePaths(argsWithPaths):
+def validateFilePath(filename, ext=None):
     """
-    This function will simply validate that the input path exists and that the output path
-    is free for the system to use
-    :param argsWithPaths: An input parsed object as provided by argparse.parse_args()
-    :return: This does not return. Simply raises ValueError in cases where paths are not valid.
+    This function will simply validate file existence and file extension.
+    If either is not true, then will raise an error.
+    :param filename: str, file path to be validated
+    :param ext: str, expected extension
+    :return: boolean True,
     """
-    gotSymbols = False
-    if not os.path.exists(argsWithPaths.input):
-        raise ValueError("ERROR: file {} does not exists.".format(argsWithPaths.input,))
+    if not os.path.exists(filename):
+        raise ValueError("ERROR: File {} does not exists.".format(filename,))
     else:
-        if os.path.exists(argsWithPaths.input.split(".")[0] + ".sym"):
-            gotSymbols = True
+        fileExt = filename.split(".")[-1]
```

Comment:
I'm not too sure how I feel about checking the file extension. Linux does not really have a concept of file extension. I think from a user standpoint that checking the file extension could be a bit annoying: what if the user made a ".backup" copy of a break point file but wants to use it? The file would be valid, would pass the existence check but would then be rejected based on the extension? I think enforcing extensions is not a good choice as it limits the user freedom.